hoomd/md/update.py

```diff
@@ -357,8 +361,122 @@ def _setattr_param(self, attr, value):
         super()._setattr_param(attr, value)
 
 
+class MeshDynamicalBonding(Updater):
+    r"""Dynamical bonding of the applied mesh that allows edge flips according to
+    a Metropolic Monte Carlo algorithm.
+
+    Args:
+        trigger (hoomd.trigger.trigger_like): Select the timesteps to triger bond
+            flip attempt.
+        mesh (hoomd.mesh.Mesh): Mesh data structure.
+        kT (float): Temperature of the simulation :math:`[\mathrm{energy}]`.
+        forces (Sequence[hoomd.md.mesh.MeshPotential]): Sequence of mesh
+          potentials applied to the updater. The default value of ``None``
+          initializes an empty list.
+
+    `MeshDynamicalBonding` works directly with `hoomd.mesh.Mesh` to apply edge
+    flips between neighboring triangles in the mesh. At each step, the updater
+    attempts to flip each edge within the mesh in random order. The probability
+    of fliping an edge :math:`ij` between the common vertices :math:`i` and
+    :math:`j` of triangles :math:`ijk` and :math:`jil` to an edge :math:`kl`
+    between :math:`k` and :math:`l` is:
+
+    .. math::
+
+        p_\mathrm{accept} =
+        \begin{cases}
+          \exp(-\beta (U(kl)-U(ij))) & U(kl) > U(ij) \\
+          1 & \Delta U(kl) \le U(ij).\\
+        \end{cases}
+
+    To obtain energies :math:`U(ij)` and :math:`U(kl)` for the corresponding edge
+    configurations, only the mesh potentials attached to the updater are
+    considered.
+
+    Tip:
+        Use `hoomd.mesh.Mesh.create_dynamical_bonding_updater` to construct a
+        `MeshDynamicalBonding` instance.
+
+    Attention:
+        `MeshDynamicalBonding` is NOT implemented for MPI parallel execution!
+
+
+    {inherited}
+
+    Examples::
+
+        mdb = hoomd.md.update.MeshDynamicalBonding(hoomd.trigger.Periodic(100),
+              mesh, kT = 1, forces=[mesh_ponetial])
+
+    """
+
+    def __init__(self, trigger, mesh, kT, forces=[]):
+        # initialize base class
+        super().__init__(trigger)
+
+        for force in forces:
+            force._allow_if_updater = True
+
```

Comment:
```suggestion
```

This seems fragile. What if the user adds another force to `updater.forces` after initializing `updater`?

I don't think this workaround is necessary anyway. See my other comment where `_allow_if_updater` is used.